cassandra/protocol_features.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from cassandra.shard_info import _ShardingInfo
+
+log = logging.getLogger(__name__)
+
+
+RATE_LIMIT_ERROR_EXTENSION = "SCYLLA_RATE_LIMIT_ERROR"
+
+class ProtocolFeatures(object):
+    rate_limit_error = None
+    shard_id = 0
+    sharding_info = None
+
+    def __init__(self, rate_limit_error=None, shard_id=0, sharding_info=None):
+        self.rate_limit_error = rate_limit_error
+        self.shard_id = shard_id
+        self.sharding_info = sharding_info
+
+    @staticmethod
+    def parse_from_supported(supported):
+        rate_limit_error = ProtocolFeatures.maybe_parse_rate_limit_error(supported)
+        shard_id, sharding_info = ProtocolFeatures.parse_sharding_info(supported)
+        return ProtocolFeatures(rate_limit_error, shard_id, sharding_info)
+
+    @staticmethod
+    def maybe_parse_rate_limit_error(supported):
+        vals = supported.get(RATE_LIMIT_ERROR_EXTENSION)
+        if vals is not None:
+            code_str = ProtocolFeatures.get_cql_extension_field(vals, "ERROR_CODE")
+            try:
+                return int(code_str)
+            except ValueError:
+                return None
```

Comment:
If this `int` fails, that means server sent us incorrect data - I think silently ignoring this is not a good choice.
Logging a warning is a minimum, but I think this warrants not catching errors here at all and letting session creation throw an error